python-backend/app/api/mines/variances/resources/variance_uploaded_documents.py

```diff
@@ -0,0 +1,30 @@
+from werkzeug.exceptions import NotFound
+
+from flask_restplus import Resource
+from app.extensions import api
+
+from ..models.variance import Variance
+from ...mine.models.mine import Mine
+from ....documents.mines.models.mine_document import MineDocument
+from ....utils.access_decorators import (requires_any_of, MINE_CREATE,
+                                         MINESPACE_PROPONENT)
+from ....utils.resources_mixins import UserMixin, ErrorMixin
+from app.api.utils.custom_reqparser import CustomReqparser
+
+
+class VarianceUploadedDocumentsResource(Resource, UserMixin, ErrorMixin):
+    @api.doc(description='Delete a document from a variance.')
+    @requires_any_of([MINE_CREATE, MINESPACE_PROPONENT])
+    def delete(self, mine_guid, variance_id, mine_document_guid):
+        variance = Variance.find_by_mine_guid_and_variance_id(mine_guid, variance_id)
+        mine_document = MineDocument.find_by_mine_document_guid(mine_document_guid)
+
+        if variance is None:
+            raise NotFound('Variance not found.')
+        if mine_document is None:
+            raise NotFound('Mine document not found.')
```

Comment:
there is no check to ensure that that document is actually attached to that variance. I believe if that was the case, it would either crash or do nothing (probably do nothing). Just thinking about our URL integrity discussion in this example.

Reply:
I had actually written a new lookup class method on `MineDocument` and used it here. But since `variance.documents.remove(mine_document)` won't find the document if it's not associated with the variance, I didn't worry about it at this time in an effort to limit the scope of this PR.

However, I did not test what the response looks like if an incorrect mine_document_guid is provided.